packages/shared/utils/__tests__/math.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { add, subtract } from '../math';
+
+describe('utils/math', () => {
+  it('should can do addition and subtraction calculation', () => {
+    expect(add(1, 2)).toBe(3);
+    expect(subtract(1, 2)).toBe(-1);
+  });
+
+  it('should fix percision issue', () => {
+    expect(add(0.1, 0.2)).toBe(0.3);
+    expect(subtract(0.3, 0.1)).toBe(0.2);
+  });
+
+  it('should can pass number-string', () => {
+    expect(add('0.1', '0.2')).toBe(0.3);
+    expect(subtract('0.3', '0.1')).toBe(0.2);
+  });
+
+  it('should be `0` when the params are `NaN`', () => {
```

Comment:
> why the title is 'should be `0` when the params are `NaN`, but test case not that ...

because:
```javascript
isNaN('string'); // true
```

I mean, if the param is **not a number**. :P